pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,61 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// BootMode is the boot mode of the system
+type BootMode string
+
+// Allowed boot mode from metal3
+const (
+	Uefi BootMode = "UEFI"
+	Legacy BootMode = "legacy"
+)
+
+// IRMCSettings is advanced BIOS settings that are supported by iRMC driver
+type  IRMCSettings struct {
+	// The UEFI FW checks the controller health status
+	EnableCheckControllersHealthStatus *bool `json:"enableCheckControllersHealthStatus,omitempty"`
+	// The number of active processor cores 1…n. Option 0 indicates that all available processor cores are active.
+	CPUActiveProcessorCores *int `json:"cpuActiveProcessorCores,omitempty"`
+	// The processor loads the requested cache line and the adjacent cache line
+	EnableCPUAdjacentCacheLinePrefetch *bool `json:"enableCPUAdjacentCacheLinePrefetch,omitempty"`
+	// The system BIOS can be written. Flash BIOS update is possible
+	EnableFlashWrite *bool `json:"enableFlashWrite,omitempty"`
+	// Boot Options will not be removed from “Boot Option Priority” list.
+	EnableKeepVoidBootOptions *bool `json:"enableKeepVoidBootOptions,omitempty"`
+	// Specifies whether the Compatibility Support Module (CSM) is executed.
+	EnableLaunchCSM *bool `json:"enableLaunchCSM,omitempty"`
+	// Prevents the OS from overruling any energy efficiency policy setting of the setup
+	EnableOSEnergyPerformanceOverride *bool `json:"enableOSEnergyPerformanceOverride,omitempty"`
+	// Active State Power Management (ASPM) is used to power-manage the PCI Express links, thus consuming less power
+	// +kubebuilder:validation:Enum=Disabled,Auto,L0Limited,L1only,L0Force
+	PciAspmSupport string `json:"pciAspmSupport,omitempty"`
+	// Specifies if memory resources above the 4GB address boundary can be assigned to PCI devices
+	EnablePciAbove4GDecoding *bool `json:"enablePciAbove4GDecoding,omitempty"`
+	// Specifies whether the switch on sources for the system are managed by the BIOS or the ACPI operating system
+	// +kubebuilder:validation:Enum=BiosControlled,AcpiControlled
+	PowerOnSource string `json:"powerOnSource,omitempty"`
+	// Single Root IO Virtualization Support is active
+	EnableSingleRootIOVirtualizationSupport *bool `json:"enableSingleRootIOVirtualizationSupport,omitempty"`
+}
+
+// BIOSConfig contains the configuration that you want to configure BIOS settings in Bare metal server
+type BIOSConfig struct {
```

Comment:
What do people think about calling this something else like BootConfig? UEFI is not BIOS.

Reply:
I don't know where the line is between BIOS and UEFI. Are the other hardware-related settings that we can configure BIOS even if the boot mode is UEFI? Are these just "hardware settings" or are they all related to how the hardware boots?

Reply:
BIOS became something of a generic term at some point during the 25 year period when it was more or less the only game in town for booting in the PC architecture. Technically BIOS and UEFI are different firmware standards for PCs, and we not only want to support both but also firmware for completely different architectures (e.g. OpenFirmware). Practically, I don't think anybody would be overly confused by BIOSConfig.
I would describe these as hardware settings that have to be configured prior to boot, i.e. in the firmware. BootConfig is not an unreasonable name, and seems preferable.

Reply:
BootConfig seems like a better and more general name - there have been various discussions also about how/where to enable other boot related things which aren't necessarily set in the bios (such as root device hints which override those hard-coded in the profiles) - those might fit OK in a BootConfig map, but would probably look odd if it's called BiosConfig?

cc @yrobla

Reply:
Based on the discussion in the metal3-docs proposal, this should be renamed "Firmware" -- https://github.com/metal3-io/metal3-docs/pull/63